libs/sdk/android/src/main/java/com/rudderstack/react/android/RNUserSessionPlugin.java

```diff
@@ -0,0 +1,206 @@
+        long timeDifference;
+        synchronized (this) {
+            timeDifference = Math.abs(currentTime - this.sessionParams.lastEventTimeStamp);
+        }
+        return timeDifference >= sessionTimeOut;
+    }
+
+    /**
+     * This saves the event time
+     */
+    void saveEventTimestamp() {
+        this.sessionParams.saveEventTimestamp();
+    }
+
+    /**
+     * This starts either a new automatic or manual session
+     */
+    void startSession() {
+        if (this.rudderClient != null) {
+            this.rudderClient.startSession();
+            RudderLogger.logVerbose("RNUserSessionPlugin: starting new session");
+        }
+    }
+
+    /**
+     * This starts a new manual session with the given session id
+     *
+     * @param sessionId The timestamp of the event
+     */
+    void startSession(long sessionId) {
+        if (this.rudderClient != null) {
+            this.rudderClient.startSession(sessionId);
+            RudderLogger.logVerbose("RNUserSessionPlugin: starting new session with sessionId: " + sessionId);
+        }
+    }
+
+    /**
+     * This ends the session
+     */
+    void endSession() {
+        if (this.rudderClient != null) {
+            this.rudderClient.endSession();
+            disableSessionParams();
+            RudderLogger.logVerbose("RNUserSessionPlugin: ending session");
+        }
+    }
+
+    /**
+     * This enables the Automatic session tracking and disables the other session tracking params
+     */
+    private void enableAutomaticSessionParams() {
+        this.sessionParams.enableSessionParams(true, false);
+    }
+
+    /**
+     * This enables the Manual session tracking and disables the other session tracking params
+     */
+    void enableManualSessionParams() {
+        this.sessionParams.enableSessionParams(false, true);
+    }
+
+    /**
+     * This disable the session tracking params
+     */
+    private void disableSessionParams() {
+        this.sessionParams.enableSessionParams(false, false);
+    }
+
+    static class SessionTrackingParams {
+        private boolean isAutomaticSessionTrackingStatus;
+        private boolean isManualSessionTrackingStatus;
+        private long lastEventTimeStamp;
+        private static RNPreferenceManager preferenceManager;
+
+        SessionTrackingParams() {
+            preferenceManager = RNPreferenceManager.getInstance();
+            refreshSessionTrackingParams();
+            this.lastEventTimeStamp = preferenceManager.getLastEventTimeStamp();
+        }
+
+        void refreshSessionTrackingParams() {
+            this.isAutomaticSessionTrackingStatus = preferenceManager.getAutomaticSessionTrackingStatus();
+            this.isManualSessionTrackingStatus = preferenceManager.getManualSessionTrackingStatus();
+        }
+
+        /**
+         * This checks whether the automatic session tracking is enabled or not
+         */
+        boolean isAutomaticSessionTrackingEnabled() {
```

Comment:
Can we rename this to something like `isAutomaticSessionTrackingActive() /  isAutomaticSessionActive` because `enabled` seems to be referring to the automatic session tracking flag set by the user by using the RudderConfig object

Reply:
The confusion seems to arrive with the usage of the same name for different purposes, refer [here](https://github.com/rudderlabs/rudder-sdk-react-native/blob/f6f9d1a34c79e341b0ba704cd9e204a386b16022/libs/sdk/android/src/main/java/com/rudderstack/react/android/RNUserSessionPlugin.java#L25)